src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -67,24 +77,34 @@ public static SqlConnection Create(IPersistentStorageFaultInjector faultInjector
 
             Contract.ThrowIfNull(handle);
 
-            raw.sqlite3_busy_timeout(handle, (int)TimeSpan.FromMinutes(1).TotalMilliseconds);
-
-            var queryToStatement = new Dictionary<string, SqlStatement>();
-            return new SqlConnection(handle, faultInjector, queryToStatement);
+            try
+            {
+                raw.sqlite3_busy_timeout(handle, (int)TimeSpan.FromMinutes(1).TotalMilliseconds);
+                return new SqlConnection(handle, faultInjector, queryToStatement);
+            }
+            catch
+            {
+                // If we failed to create connection, ensure that we still release the sqlite
+                // handle.
+                raw.sqlite3_close(handle);
+                throw;
+            }
         }
 
         private SqlConnection(sqlite3 handle, IPersistentStorageFaultInjector faultInjector, Dictionary<string, SqlStatement> queryToStatement)
         {
-            // This constructor avoids allocations since failure (e.g. OutOfMemoryException) would leave the object
-            // partially-constructed, and the finalizer would run later triggering a crash.
+            // This constructor avoids allocations since failure (e.g. OutOfMemoryException) would
+            // leave the object partially-constructed, and the finalizer would run later triggering
+            // a crash.
```

Comment:
💡 I already wrapped this to our standard of 120 columns. I'd leave it unless you are needing to change content as well.